### ISSUE TITLE
Login problem on Medium based blogs

### ISSUE THREAD

**Issue by jaydson** [brave/browser-laptop#5499]
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
It seems like Medium is always signing me off on Brave.

**Expected behaviour:**
Login on Medium, keep logged in.

- Platform (Win7, 8, 10? macOS? Linux distro?):
Ubuntu 16.10

- Brave Version:
0.12.7

- Steps to reproduce:
    1. Navigate to https://thisweekinreact.com (or any other Medium based blog)
    2. Login
    3. Reload the page

- Screenshot if needed:
![screenshot from 2016-11-08 22-51-51](https://cloud.githubusercontent.com/assets/72206/20123469/7401aa3c-a606-11e6-95d3-7789628960df.png)

- Any related issues:
#5437

**Comment by anasant** [comment]
Tried to look into this. Can't reproduce.

**Comment by bsclifton** [comment]
+1 from Community:
https://community.brave.com/t/medium-com-publications-cause-sign-out/2985/3

Steps to reproduce:
* Allow all cookies for Medium.com
* Allow all cookies for https://themission.co (this is a publication connected to Medium)
* Sign into Medium.com, you'll see your profile is active by looking at the top right-hand corner
* Click on any article on The Mission (Like this one)
* Click the back button or return to Medium.com 
* You'll now be signed out of Medium

**Comment by jamesray1** [comment]
Yes I can also reproduce.

https://blog.gridplus.io/why-ethereum-needs-stable-coins-b777b55945b5

**Comment by alex-miller-0** [comment]
I can reproduce this. 

Brave: 0.19.88 
rev: c29eca63249e3d43584ca1dcc47f3e7a7e904324 
Muon: 4.5.15 
libchromiumcontent: 62.0.3202.89 
V8: 6.2.414.40 
Node.js: 7.9.0 
Update Channel: Release 
OS Platform: macOS 
OS Release: 16.7.0 
OS Architecture: x64

**Comment by jonathansampson** [comment]
@alex-miller-0 Does this repro for you if you lower your shields for that domain?

**Comment by alex-miller-0** [comment]
Yes, shields are down for medium.com (and all relevant domains for blogs). It appears to have more to do with the tab itself: 

* If I open a new tab and navigate to medium, I am not signed in. 
* If I open a new story in a new tab (i.e. click a link) it signs me out on all tabs.
* Interestingly, if I navigate inside the existing tab (i.e. click on the M logo to return to medium.com), I remain signed in. If I click on a story, I'm still signed in, but once I click the M logo a second time, I get logged out.

**Comment by jamesray1** [comment]
When I clicked on the Medium logo while signed in I was signed out.

When trying to add tags, as soon as I type a comma or two spaces, it repeatedly says after every keyboard after that "Uh oh!
Tags can only have letters, numbers, spaces, and dashes, and limited to 5 per post."

**Comment by jaydson** [comment]
I'm having a different and weird behaviour here.  
I can login on Medium and navigate through different posts.  
Steps:  
1) Go to https://medium.com  
2) Login with Twitter  
3) Navigate to a personal Medium account, like https://medium.com/@jaydson/  
4) Navigate  through different posts
This steps works just fine.

But the odd thing is, in the moment I access a custom domain Medium based blog, like https://thisweekinreact.com, I'm signed out. 

Bottom line is, Medium personal profiles seems to work fine.
Medium with custom domains keep signing me out.

**Comment by superMDguy** [comment]
Yeah, it's definitely related to custom domains for me as well. Whenever I go to a page on a custom domain, I get logged out on the main site.

**Comment by alex-miller-0** [comment]
Ah, it is related to custom domains for me too. I have brought down shields on the domains I was testing, but the bug only happens when I leave the medium domain for a custom domain.

**Comment by Shad-Iqbal** [comment]
And it's still happening... No **solution** found!

**Comment by jpita** [comment]
this is still in deep backlog?

**Comment by jgmio** [comment]
Same issue as described above

**Comment by jonathansampson** [comment]
@Shad-Iqbal, @jpita, @jgmio--would one of you be interested in joining me via Skype or Discord to have a closer look at this issue?

**Comment by Liunkae** [comment]
I ran into this problem as well. I think I worked around it by setting the global default shields Cookie Control to Allow all cookies. The problem is probably because of third party cookies (maybe medium.com tries to pass cookies to the custom domain?).

**Comment by jpita** [comment]
@jonathansampson sure. jose.pita4 on skype

**Comment by j1v** [comment]
I have this problem since day one with medium on brave. 

It makes me login to medium dozens of times a day

**Comment by nneuman** [comment]
+1 on this - is it partly due to custom domains pointing to Medium publications? 

I wonder if Medium uses a tag of some sort that could identify publications even if they're using a custom domain?

**Comment by Liunkae** [comment]
+1 from https://medium.com/@PaulRBerg/constructive-feedback-for-the-brave-browser-9f08b422a05

Like I said above, as a workaround, please try changing global default shields Cookie Control to Allow all cookies.

![](https://user-images.githubusercontent.com/19424103/41562309-d5c6fba8-7311-11e8-9494-8ad1c3e7847d.png)

**Comment by jgmio** [comment]
But that completely defeats the purpose of the brave browser.
-1 for this one

**Comment by nneuman** [comment]
I agree with @jgmio, no reason to use Brave if I have to turn off the security features. I've gone back to Chrome + a variety of privacy extensions.

**Comment by Liunkae** [comment]
@jgmio @nneuman This is just a workaround until the underlying problem is fixed. You can also reset all cookies on browser quit.
![](https://user-images.githubusercontent.com/19424103/41606989-9c774152-73aa-11e8-9660-da6128f63f2b.png)
We'll also have to see if the new version of Brave fixes this problem for good. https://brave.com/development-plans-for-upcoming-release/

**Comment by snyderp** [comment]
I'm not able to recreate in the beta
